providers/openfeature-go-feature-flag-provider/lib/openfeature/go-feature-flag/go_feature_flag_provider.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+module OpenFeature
+  module GoFeatureFlag
+    # This class is the entry point for the GoFeatureFlagProvider
+    class Provider
+      PROVIDER_NAME = "GO Feature Flag Provider"
+      attr_reader :metadata, :options
+
+      def initialize(options: OpenFeature::GoFeatureFlag::Options.new)
+        @metadata = OpenFeature::SDK::Provider::ProviderMetadata.new(name: PROVIDER_NAME)
+        @options = options
+        @goff_api = OpenFeature::GoFeatureFlag::GoFeatureFlagApi.new(options: options)
+      end
+
+      def fetch_boolean_value(flag_key:, default_value:, evaluation_context: nil)
+        evaluate(flag_key: flag_key, default_value: default_value, evaluation_context: evaluation_context, allowed_classes: [TrueClass, FalseClass])
+      end
+
+      def fetch_string_value(flag_key:, default_value:, evaluation_context: nil)
+        evaluate(flag_key: flag_key, default_value: default_value, evaluation_context: evaluation_context, allowed_classes: [String])
+      end
+
+      def fetch_number_value(flag_key:, default_value:, evaluation_context: nil)
+        evaluate(flag_key: flag_key, default_value: default_value, evaluation_context: evaluation_context, allowed_classes: [Integer, Float])
+      end
+
+      def fetch_object_value(flag_key:, default_value:, evaluation_context: nil)
+        evaluate(flag_key: flag_key, default_value: default_value, evaluation_context: evaluation_context, allowed_classes: [Array, Hash])
+      end
+
+      private
+
+      def evaluate(flag_key:, default_value:, allowed_classes:, evaluation_context: nil)
+        evaluation_context = OpenFeature::SDK::EvaluationContext.new unless evaluation_context.is_a?(OpenFeature::SDK::EvaluationContext)
```

Comment:
I don't exactly get the check on type specifically

```suggestion
        evaluation_context = OpenFeature::SDK::EvaluationContext.new if evaluation_context.nil?
```